polaris-react/src/components/Button/Button.tsx

```diff
@@ -34,26 +32,22 @@ export interface ButtonProps extends BaseButton {
   size?: 'micro' | 'slim' | 'medium' | 'large';
   /** Changes the inner text alignment of the button */
   textAlign?: 'left' | 'right' | 'center' | 'start' | 'end';
-  /** Gives the button a subtle alternative to the default button styling, appropriate for certain backdrops */
-  outline?: boolean;
   /** Allows the button to grow to the width of its container */
   fullWidth?: boolean;
   /** Displays the button with a disclosure icon. Defaults to `down` when set to true */
   disclosure?: 'down' | 'up' | 'select' | boolean;
-  /** Renders a button that looks like a link */
-  plain?: boolean;
-  /** Makes `plain` and `outline` Button colors (text, borders, icons) the same as the current text color. Also adds an underline to `plain` Buttons */
-  monochrome?: boolean;
-  /** Removes underline from button text (including on interaction) when `monochrome` and `plain` are true */
+  /** Removes underline from button text (including on interaction) */
   removeUnderline?: boolean;
   /** Icon to display to the left of the button content */
   icon?: React.ReactElement | IconSource;
   /** Disclosure button connected right of the button. Toggles a popover action list. */
   connectedDisclosure?: ConnectedDisclosure;
   /** Indicates whether or not the button is the primary navigation link when rendered inside of an `IndexTable.Row` */
   dataPrimaryLink?: boolean;
-  /** Extra visual weight combined with indication of a positive action */
-  primarySuccess?: boolean;
+  /** Sets the color treatment of the Button. */
+  tone?: 'critical' | 'success';
+  /** Changes the visual appearance of the Button. */
+  variant?: 'plain' | 'primary' | 'primaryPlain' | 'monochromePlain';
```

Comment:
Great points @samrose3. I agree the variants are less than ideal at the moment. The `primaryPlain` and `monochromePlain` are hacky workarounds for the state of things at the moment.

`tone` I am trying to follow [the API guidelines](https://github.com/Shopify/ui-api-design/tree/main/codex#the-different-design-variations) and my understanding is it will show up in other components like `Banner` so it would make sense to keep the values consistent between components. `critical` and `success` make sense applied to `Banner` but less sure about `tone="primary"` for `Banner`? `primary` seems more like a `Button` style to me.

Reply:
> I like those ideas @samrose3 if we were planning on keeping plainMonochrome but I would hesitate to introduce a neutral tone for a variant that is a hack and really should be removed.

➕ 

![image](https://github.com/Shopify/polaris/assets/8629173/3098445a-fc9d-4632-9709-a785d436beaa)
This I would expect to be `tertiary` with inverted colors

![image](https://github.com/Shopify/polaris/assets/8629173/5dfd7ddd-c66b-4452-90ae-8405b9ab54ae)
This I would expect to be `tertiary`

Some of this is tough because with these updates we also have to account for the space they take up so I don't expect all these changes to happen now. Just some examples to Aveline's point of `plainMonochrome` isn't a long term pattern we want to promote.

Reply:
>This I would expect to be tertiary with inverted colors

So does this become a `tertiaryInverted` variant? Or how do we propose handling inverted? It's probably something that could apply to all variants?

This is something we need to put more thought into and until we have it a bit more solidified at the system level, `plainMonochrome` is the iterim state.

Reply:
I agree